src/analyzer/analyzerthread.cpp

```diff
@@ -249,36 +249,45 @@ AnalyzerThread::AnalysisResult AnalyzerThread::analyzeAudioSource(
                         mixxx::WritableSampleFrames(
                                 inputFrameIndexRange,
                                 mixxx::SampleBuffer::WritableSlice(m_sampleBuffer)));
+        DEBUG_ASSERT(readableSampleFrames.frameIndexRange() <= inputFrameIndexRange);
+        // Sometimes the duration of the audio source is inaccurate
+        // and adjusted (= shrinked) while reading.
+        remainingFrames = intersect(remainingFrames, audioSourceProxy.frameIndexRange());
+        inputFrameIndexRange = intersect(inputFrameIndexRange, audioSourceProxy.frameIndexRange());
+        DEBUG_ASSERT(readableSampleFrames.frameIndexRange() <= inputFrameIndexRange);
 
         sleepWhileSuspended();
         if (isStopping()) {
             return AnalysisResult::Cancelled;
         }
 
         // 2nd: step: Analyze chunk of decoded audio data
-        if (readableSampleFrames.frameLength() == mixxx::kAnalysisFramesPerBlock ||
-                remainingFrames.empty()) {
-            // Complete chunk of audio samples has been read for analysis
-            for (auto&& analyzer : m_analyzers) {
-                analyzer.processSamples(
-                        readableSampleFrames.readableData(),
-                        readableSampleFrames.readableLength());
-            }
-            if (remainingFrames.empty()) {
-                result = AnalysisResult::Complete;
-            }
+        for (auto&& analyzer : m_analyzers) {
+            analyzer.processSamples(
+                    readableSampleFrames.readableData(),
+                    readableSampleFrames.readableLength());
+        }
+
+        // Check if complete or abort on errors
+        if (remainingFrames.empty()) {
+            result = AnalysisResult::Complete;
         } else {
-            // Partial chunk of audio samples has been read, but not the final.
-            // A decoding error must have occurred, maybe a corrupt file?
-            kLogger.warning()
-                    << "Aborting analysis after failure to read sample data:"
-                    << "expected frames =" << inputFrameIndexRange
-                    << ", actual frames =" << readableSampleFrames.frameIndexRange();
-            result = AnalysisResult::Partial;
+            // Only the final chunk should be incomplete
```

Comment:
Is this comment correct? I think not, because even the final chunk is clammed by the total length.